src/core/exception.d

```diff
@@ -47,24 +47,34 @@ else:
  */
 class RangeError : Error
 {
-    @safe pure nothrow this( string file = __FILE__, size_t line = __LINE__, Throwable next = null )
+    @safe pure nothrow this( string file = __FILE__, size_t line = __LINE__, Throwable next = null)
+    {
+        this( 0, 0, 0, file, line, next);
+    }
+    @safe pure nothrow this( size_t lwr, size_t upr, size_t len,
+                             string file = __FILE__, size_t line = __LINE__, Throwable next = null)
     {
         super( "Range violation", file, line, next );
+        lower = len;
+        upper = upr;
+        length = len;
     }
+    size_t lower, upper; // lower/upper bound passed, if slice (array[lower .. upper]), if not a slice (array[upper])
+    size_t length;       //length of the indexed array
```

Comment:
* Instance variables should go before an methods/constructors
* `const`/`immutable` as well?